### DISCUSSION TITLE
Analytics et anonymisation

### DISCUSSION THREAD

**Discussion by laudag** [AmauriC/tarteaucitron.js#692] [Q&A]
Salut !
j'ai un problème pour implanter le boit de code d'anonymisation d'Analytics avec GTAG. Malgré des recherche approfondies sur ce truc qui me semblait classique, impossible de faire ressortir quelque chose de clair. J'ai écrit à rdorian du plugin WP auto hébergé de tarteaucitron qui m'a recommandé de venir poser la question ici.  j’utilise le code de suivi Analytics suivant : 

<script async src="https://www.googletagmanager.com/gtag/js?id=XX-XXXXXXX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'XX-XXXXXXX');
</script>

Est ce le Google Analytics (universal), le Google Analytics (ga.js) ou le Google Analytics (gtag.js) ?

Et aussi, plus important, mais peut*-être lié : je ne sais pas si l’anonymisation est déjà dans le script de tarteaucitron ou s’il faut l’ajouter dans la zone fonction du code JS de tarteaucitron à implanter dans les pages. Dans ce cas, impossible de trouver une ressource donc pour savoir comment exactement mettre ce bout de code pour anonymiser les stats, genre
tarteaucitron.user.analyticsMore = function () { gatg(‘set’, ‘anonymizeIp’, true); };
Mais je trouve de tout et surtout avec ga et pas gatg.
Une idée de ce qu’il faut mettre si ce n’est pas par défaut dans le script tarteaucitron ?

Grand merci d’avance
Laurent

**Comment by AmauriC** [comment]
Hello

C'est le service gtag.js pour ce marqueur.

Pour anonymiser les ips il faut bien ajouter:

`tarteaucitron.user.analyticsMore = function () { gatg('set', 'anonymizeIp', true); };`

**Reply by laudag** [reply]
Superbe ! 
Merci pour ta réponse rapide.

**Comment by laudag** [comment]
En fait ça ne fonctionne pas. Même en remplaçant tarteaucitron.user.analyticsMore par tarteaucitron.user.gatgMore ce qui semblait une erreur de ma part.
En revanche, en allant fafouiller dans le code JS de tarteaucitron 1.9.1, la dernière version je pense, il semblerait que le script du service gatg comporte la fonction d'anonymisation. Je me trompe ?

**Comment by AmauriC** [comment]
Exact `'anonymize_ip': true` est présent dans le tag tarteaucitron : )
